### DISCUSSION TITLE
Consumption numbers in the Trip tab inconsistent with those reported by Tesla UI

### DISCUSSION THREAD

**Discussion by tanglewood16** [teslamate-org/teslamate#2899] [General]
Hi, I have noticed that the consumption numbers reported in the Teslamate Trip tab are greater than those numbers reported by the Tesla UI, typical in the range of 5% to 10% more. Here's an example:

![image](https://user-images.githubusercontent.com/47452148/190648454-31678b9f-6804-4947-8ec6-72f54daf0839.png)

I made a roundtrip this morning. Tesla UI reported 218wh/mi for the trip. Teslamate broke the trip into two segments (I did shift the car into Park at the destination for a brief stop, but didn't get off my seat), and the average of the two segments was 234wh/mi, ~ 7% more.

Any insight on the reason for the discrepancy? Thanks!

**Comment by Esparza13** [comment]
Hi, I have noticed exactly the same. 
Crossing data with Scanmytesla it seems that teslamate takes the "nominal full pack" instead of the usable capacity taking out the energy buffer. For me this is 75 kWh vs 71.5 kWh, around 5 % difference. This means that teslamate takes the spent SoC*75 instead of spentSoC*71.5 to get the energy and then calculate the consumption. 

But I dont know how to fix it.

**Comment by vexingly** [comment]
According to the faq (https://github.com/adriankumpf/teslamate/blob/master/website/docs/faq.md) it looks like this is actually using a custom value based on charging data... which should technically be more accurate than using a static value of the pack size and account for degradation.

```
Why are no consumption values displayed in Grafana?
Unfortunately the Tesla API does not return consumption values for a trip. In order to still be able to display values TeslaMate estimates the consumption on the basis of the recorded (charging) data. It takes at least two charging sessions before the first estimate can be displayed. Charging sessions have to be longer than 10 minutes and less than 95% state-of-charge (SoC). Each future charging session will slightly improve the accuracy of the estimate, which is applied retroactively to all data.
```

**Reply by lq313c** [reply]
Are you saying that Teslamate's method for measuring consumption is more accurate than what the car's UI shows? I have no idea how the car measures consumption either so I don't know for sure, but I've always assumed that it was directly measuring current drawn.

**Reply by Esparza13** [reply]
So I guess what we get is the consumption including the charging losses. 
Maybe it would be possible to have it also without them. The typical range variation multiplied by the car typical consumption will give us the spent energy from the battery.

**Reply by vexingly** [reply]
I would agree that Tesla is probably measuring consumption directly and is the most accurate. 

The TeslaMate charging tab does have the energy Added and energy Used, as well as the battery %, so using that info from the car directly it should be possible to get a pretty good estimate of the battery capacity available... but it is still an estimate, hopefully it becomes more accurate over time.

For short trips it would probably be overestimated, unless the battery percentage has a higher precision than it seems and/or there is a manual correction... i.e. if you start your drive at 37% maybe it is actually 37.9% or 37.1%.
